bundle/edu.gemini.spModel.smartgcal/src/main/java/edu/gemini/spModel/smartgcal/provider/CalibrationProviderImpl.java

```diff
@@ -33,18 +33,30 @@ public CalibrationProviderImpl(CalibrationRepository repository) {
 
     @Override
     public List<Calibration> getCalibrations(CalibrationKey key) {
-        ConfigurationKey configKey = key.getConfig();
+        final ConfigurationKey configKey = key.getConfig();
+        final List<Calibration> cals;
         if (key instanceof CalibrationKeyImpl.WithWavelength) {
-            Double wavelength = ((CalibrationKeyImpl.WithWavelength) key).getWavelength();
-            return getCalibrations(
-                    configKey.getInstrumentName(),
-                    configKey,
-                    wavelength);
+            final Double wavelength = ((CalibrationKeyImpl.WithWavelength) key).getWavelength();
+            cals = getCalibrations(configKey.getInstrumentName(), configKey, wavelength);
         } else {
-            return getCalibrations(
-                    configKey.getInstrumentName(),
-                    key.getConfig());
+            cals = getCalibrations(configKey.getInstrumentName(), key.getConfig());
         }
+
+        // Partition the calibrations into flats and arcs.
+        final List<Calibration> flats = new ArrayList<>();
+        final List<Calibration> arcs  = new ArrayList<>();
+        cals.forEach( cal -> {
+            if (cal.isFlat()) {
+                flats.add(cal);
+            } else {
+                arcs.add(cal);
+            }
+        });
+
+        // Return the flats before the arcs.
+        final List<Calibration> result = new ArrayList<>(flats);
+        result.addAll(arcs);
+        return result;
```

Comment:
This is the heart of the thing.  There are more Java-8ish ways to do this but I find them equally awkward.  Dunno.

This one may be clearer.  Makes several passes through the calibrations.
````java
        final Stream<Calibration> calStream  = cals.stream();
        final Stream<Calibration> flatStream = calStream.filter(c -> c.isFlat());
        final Stream<Calibration> arcStream  = calStream.filter(c -> c.isArc());
        return Stream.concat(flatStream, arcStream).collect(Collectors.toList());
````

Then there's the awkward `Collectors`:

````java
        final Map<Boolean, List<Calibration>> flatsArcs =
                cals.stream()
                    .collect(Collectors.partitioningBy(c -> c.isFlat()));
        
        final List<Calibration> result = new ArrayList(flatsArcs.get(true));
        result.addAll(flatsArcs.get(false));
        return result;
````

I'm inclined to leave it as is unless there's a better way?